staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -192,7 +192,13 @@ func newETCD3Storage(c storagebackend.Config) (storage.Interface, DestroyFunc, e
 		// TODO: fix duplicated storage destroy calls higher level
 		once.Do(func() {
 			stopCompactor()
-			client.Close()
+			go func() {
+				select {
+				case <-time.After(5 * time.Second):
+					// Grace period for in-flight requests
+					client.Close()
+				}
+			}()
```

Comment:
I'll remove this once https://github.com/kubernetes/kubernetes/pull/77816 merges and I rebase on it

Reply:
It merged :)